### PR TITLE
Fix: Form tsd

### PR DESCRIPTION
**Pull request by paranoidjk**
Ref: https://github.com/react-component/form/blob/master/src/createBaseForm.js#L499

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: components/_util/type.ts [hunk 1]
```diff
@@ -1,3 +1,2 @@
-export type Diff<T extends string, U extends string> = ({ [P in T]: P } &
-  { [P in U]: never } & { [x: string]: never })[T];
-export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
+// https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#predefined-conditional-types
+export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
```
[the remark this answers is not in the document]
Comment (paranoidjk) [reply components/_util/type.ts]: 这个可改可不改，不影响运行时

[the remark this answers is not in the document]
Comment (paranoidjk) [reply components/_util/type.ts]: 主要是修类型定义的问题。snapshot 挂了是什么模块更新搞挂了？

[the remark this answers is not in the document]
Comment (yesmeck) [reply components/_util/type.ts]: rebase 下吧。

还是不要改了，少点 surprise.

[the remark this answers is not in the document]
Comment (paranoidjk) [reply components/_util/type.ts]: done

**Comment by icodytan** [comment]
@paranoidjk Can you fix the problem of @Form.create()? [proposed to modify the form.d.ts](https://github.com/ant-design/ant-design/issues/9874)

**Pull request merged**